akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -154,15 +156,48 @@ import scala.util.control.NonFatal
   }
 
   override def postStop(): Unit = {
-    behavior match {
-      case null                   ⇒ // skip PostStop
-      case _: DeferredBehavior[_] ⇒
-      // Do not undefer a DeferredBehavior as that may cause creation side-effects, which we do not want on termination.
```

Comment:
this feels scary to change for the reason mentioned in this comment, e.g. spawning a child

Reply:
should we perhaps throw immediately if `Behavior.stopped(postStop)` is used with a `DeferredBehavior` as the `postStop`?